executor/src/witgen/processor.rs

```diff
@@ -239,7 +241,14 @@ Known values in current row (local: {row_index}, global {global_row_index}):
                 }
                 error += &format!("   => Error: {e}");
                 error.into()
-            })?;
+            });
+        let updates = match updates {
+            Ok(updates) => updates,
+            Err(e) => {
+                record_end_identity(identity.id);
+                return Err(e);
+            }
+        };
```

Comment:
Ah nooo ^^

I debugged the error I mentioned in the other comment, and it's actually not because of that but because there is an `record_end_identity(identity.id);` missing before the return in 256...